FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : Value
+    {
+        public string Id { get; }
+        public RefV Class { get; }
+        public RefV Database { get; }
+
+        public RefV(string id, RefV @class = null, RefV database = null)
+        {
+            Id = id;
+            Class = @class;
+            Database = database;
+
+            if (@class != null && !@class.IsClassRef())
```

Comment:
I understand that the developer could build a wrong ref without this validations but, I think this should be the db's responsibility.